src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Deployments/StopAzureDeploymentCmdlet.cs

```diff
@@ -0,0 +1,74 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Management.Automation;
+using Microsoft.Azure.Commands.ResourceManager.Cmdlets.Components;
+using Microsoft.Azure.Commands.ResourceManager.Cmdlets.SdkModels;
+using ProjectResources = Microsoft.Azure.Commands.ResourceManager.Cmdlets.Properties.Resources;
+
+namespace Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation
+{
+    /// <summary>
+    /// Cancel a running deployment.
+    /// </summary>
+    [Cmdlet(VerbsLifecycle.Stop, "AzureRmDeployment", SupportsShouldProcess = true,
+        DefaultParameterSetName = StopAzureDeploymentCmdlet.DeploymentNameParameterSet), OutputType(typeof(bool))]
+    public class StopAzureDeploymentCmdlet : ResourceManagerCmdletBase
+    {
+        /// <summary>
+        /// The deployment Id parameter set.
+        /// </summary>
+        internal const string DeploymentIdParameterSet = "StopByDeploymentId";
+
+        /// <summary>
+        /// The deployment name parameter set.
+        /// </summary>
+        internal const string DeploymentNameParameterSet = "StopByDeploymentName";
+
+        /// <summary>
+        /// The input object parameter set.
+        /// </summary>
+        internal const string InputObjectParameterSet = "StopByInputObject";
+
+        [Alias("DeploymentName")]
+        [Parameter(Position = 0, ParameterSetName = StopAzureDeploymentCmdlet.DeploymentNameParameterSet,
+            Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The name of the deployment.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Alias("DeploymentId", "ResourceId")]
+        [Parameter(ParameterSetName = StopAzureDeploymentCmdlet.DeploymentIdParameterSet, Mandatory = true,
```

Comment:
@Tiano2017 same comment about adding the `ValueFromPipelineByPropertyName=true` property to this parameter and removing it from `-Name`

Reply:
will keep this for the same reason. :)

Reply:
@Tiano2017 same comment as above 😀